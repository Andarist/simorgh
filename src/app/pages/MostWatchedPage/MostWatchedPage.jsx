import React, { useContext } from 'react';
import styled from 'styled-components';
import { node } from 'prop-types';
import {
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MAX,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MAX,
  GEL_GROUP_5_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/breakpoints';
import {
  GEL_SPACING_TRPL,
  GEL_SPACING_QUAD,
} from '@bbc/gel-foundations/spacings';
import { ServiceContext } from '#contexts/ServiceContext';
import Grid, { GelPageGrid } from '#app/components/Grid';
import IndexMain from '#app/components/PageLayout/IndexMain';
import IndexPageContainer from '#app/components/PageLayout/IndexPageContainer';
import IndexHeading from '#containers/IndexHeading';

const MarginWrapper = styled.div`
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
    margin-top: ${GEL_SPACING_TRPL};
  }

  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
    margin-top: ${GEL_SPACING_QUAD};
  }

  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    margin-top: ${GEL_SPACING_QUAD};
  }
`;

const MostWatchedPage = () => {
  const {
    dir,
    mostWatched: { header },
  } = useContext(ServiceContext);

  const MostWatchedWrapper = ({ children }) => (
    <>
      <IndexHeading id="content">{header}</IndexHeading>
      <MarginWrapper>
        <GelPageGrid
          dir={dir}
          columns={{
            group0: 6,
            group1: 6,
            group2: 6,
            group3: 6,
            group4: 8,
            group5: 20,
          }}
          enableGelGutters
        >
          <Grid
            item
            dir={dir}
            startOffset={{
              group0: 1,
              group1: 1,
              group2: 1,
              group3: 1,
              group4: 1,
              group5: 3,
            }}
            columns={{
              group0: 6,
              group1: 6,
              group2: 6,
              group3: 6,
              group4: 6,
              group5: 11,
            }}
          >
            {children}
          </Grid>
        </GelPageGrid>
      </MarginWrapper>
    </>
  );

  MostWatchedWrapper.propTypes = {
    children: node.isRequired,
  };

  return (
    <>
      <IndexMain data-e2e="most-watched">
        <IndexPageContainer>
          <MostWatchedWrapper>
            <div style={{ display: 'none' }}>
              To be replaced with most watched content
            </div>
          </MostWatchedWrapper>
        </IndexPageContainer>
      </IndexMain>
    </>
  );
};

export default MostWatchedPage;
