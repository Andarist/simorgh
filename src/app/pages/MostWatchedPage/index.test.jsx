import React from 'react';
import { render, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { matchSnapshotAsync } from '@bbc/psammead-test-helpers';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import MostWatchedPage from './MostWatchedPage';

const MostWatchedPageWithContext = () => (
  <ServiceContextProvider service="pidgin">
    <RequestContextProvider
      bbcOrigin="https://www.test.bbc.com"
      isAmp={false}
      pageType="mostWatched"
      pathname="/pathname"
      service="pidgin"
      statusCode={200}
    >
      <BrowserRouter>
        <MostWatchedPage />
      </BrowserRouter>
    </RequestContextProvider>
  </ServiceContextProvider>
);

describe('Most Watched Page Main', () => {
  it('should match snapshot for most watched page', async () => {
    await matchSnapshotAsync(<MostWatchedPageWithContext />);
  });

  it('shoulder render most watched page', async () => {
    let container;
    await act(async () => {
      container = await render(<MostWatchedPageWithContext />).container;
    });

    expect(container.querySelector('h1').textContent).toEqual(
      'De one we dem don look',
    );
  });
});
