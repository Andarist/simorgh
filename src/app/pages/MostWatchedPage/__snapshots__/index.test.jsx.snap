// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Most Watched Page Main should match snapshot for most watched page 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  margin: 0 0.5rem;
  padding-top: 0.5rem;
  padding-bottom: 2rem;
}

.c2 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #6E6E73;
  margin: 0;
  padding-bottom: 1.5rem;
}

@supports (display:grid) {
  .c4 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c6 {
    width: calc(6/6*(100% - 6 * 0.5rem) + 5 * 0.5rem - 0%);
    margin: 0 0.25rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c6 {
    width: calc(6/6*(100% - 6 * 0.5rem) + 5 * 0.5rem - 0%);
    margin: 0 0.25rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c6 {
    width: calc(6/6*(100% - 6 * 0.5rem) + 5 * 0.5rem - 0%);
    margin: 0 0.25rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c6 {
    width: calc(6/6*(100% - 6 * 1rem) + 5 * 1rem - 0%);
    margin: 0 0.5rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:79.9375rem) {
  .c6 {
    width: calc(6/8*(100% - 8 * 1rem) + 5 * 1rem );
    margin: 0 0.5rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c6 {
    width: calc(11/20*(100% - 20 * 1rem) + 10 * 1rem );
    margin: 0 0.5rem;
    margin-left: 10%;
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c6 {
    display: block;
  }
}

@media (min-width:63rem) and (max-width:79.9375rem) {
  .c5 {
    margin: 0 auto;
    max-width: 63rem;
  }
}

@media (min-width:80rem) {
  .c5 {
    margin: 0 auto;
    max-width: 80rem;
  }
}

@media (min-width:25rem) {
  .c1 {
    margin: 0 1rem;
  }
}

@media (min-width:25rem) {
  .c1 {
    padding-top: 1rem;
  }
}

@media (max-width:37.4375rem) {
  .c1 {
    padding-bottom: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    padding-top: 0;
  }
}

@media (min-width:63rem) {
  .c1 {
    padding-bottom: 2.5rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c2 {
    padding: 1.5rem 0 0.5rem;
  }
}

@media (min-width:63rem) {
  .c2 {
    padding: 1.5rem 0 0;
  }
}

@media (min-width:80rem) {
  .c2 {
    width: 100%;
    margin: 0 auto;
    max-width: 80rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c3 {
    margin-top: 1.5rem;
  }
}

@media (min-width:63rem) and (max-width:79.9375rem) {
  .c3 {
    margin-top: 2rem;
  }
}

@media (min-width:80rem) {
  .c3 {
    margin-top: 2rem;
  }
}

<main
  class="c0"
  data-e2e="most-watched"
  role="main"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
      dir="ltr"
      id="content"
      tabindex="-1"
    >
      De one we dem don look
    </h1>
    <div
      class="c3"
    >
      <div
        class="c4 c5"
        dir="ltr"
      >
        <div
          class="c6"
          dir="ltr"
        >
          <div
            style="display: none;"
          >
            To be replaced with most watched content
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
